### PR TITLE
instructorHelp: Update UI based examples #7264

### LINKED ISSUE #7264: instructorHelp: Update UI based examples
The UI based examples are outdated. Update them according to the latest UI. Eg. this is the example given in `instructorHelp.jsp`, for `Sessions -> Setting up questions`.

This is the actual UI the instructor sees when creating a new essay question -

Make such updates wherever necessary.

### PR DESCRIPTION
**Pull request by kguo901**
Fixes #7264

### CODE REVIEW

**Comment by kguo901** [comment]
This issue is ongoing. 
I made changes to a single example first as requested by @damithc. See below for a screenshot of the result. If this change is approved, I will continue to update the rest of the examples on the instructor help page.
uichange

File: src/main/webapp/instructorHelp.jsp [hunk 1]
```diff
@@ -5329,7 +5113,7 @@
                                               </ul>
                                           </div>
                                           <div class="feedback-path-others margin-top-7px" style="display:none;">
-                                              <div data-original-title="Who will give feedback" class="col-sm-12 col-lg-6 padding-0 margin-bottom-7px" data-toggle="tooltip" data-placement="top" title="">  
+                                              <div data-original-title="Who will give feedback" class="col-sm-12 col-lg-6 padding-0 margin-bottom-7px" data-toggle="tooltip" data-placement="top" title="">
```
Comment (thenaesh) [inline src/main/webapp/instructorHelp.jsp]: I'm assuming these are unintentional. The only thing done is removing a trailing space. Admittedly, they shouldn't have been there in the first place, but each PR should do only one thing.

File: src/main/webapp/instructorHelp.jsp [hunk 2]
```diff
@@ -8329,11 +8113,11 @@
                                                 <span>
                                                     <strong>Question</strong>
                                                     <select class="questionNumber nonDestructive text-primary" name="questionnum" id="questionnum-4" disabled="">
-                                                        
+
```
Comment (thenaesh) [inline src/main/webapp/instructorHelp.jsp]: These too.

File: src/main/webapp/instructorHelp.jsp (hunk 1 above)
Comment (kguo901) [inline src/main/webapp/instructorHelp.jsp]: Sorry about these extraneous changes. The way I was editing the document must have modified the spacing. I will make the changes soon.

**Comment by kguo901** [comment]
I re-added the whitespaces. Please let me know if I missed any.

**Comment by thenaesh** [comment]
Looks ok so far. For your subsequent commits, please make sure that every commit makes exactly one logical change and contains no extraneous changes.

Tip: `git commit --amend --no-edit` can be used to add changes into your latest commit without creating an entirely new commit or manually rebasing.

**Comment by kguo901** [comment]
@thenaesh Thanks for the tip!

**Comment by kguo901** [comment]
@thenaesh Some UI changes require me to add additional dropdowns. However, dropdowns like the one for time zones in the "create a course" example have an extremely large number of options. 
2017-05-21_li
Including all of these options in instructorHelp.jsp would cause my local server to crash.  What is your suggesting on how to go about this? Should I only include the first 10 or so options?

**Comment by damithc** [comment]
> Including all of these options in instructorHelp.jsp would cause my local server to crash. What is your suggesting on how to go about this? Should I only include the first 10 or so options?

yes, you can include a few options with a `Other options omitted...`

**Comment by whipermr5** [comment]
@kguo901 Is this ready for review? :)

**Comment by kguo901** [comment]
@whipermr5 Not yet, there are still several updates I need to make. I will try to finish everything by the end of this week.

**Comment by kguo901** [comment]
Ready for review!

**Comment by whipermr5** [comment]
@kguo901 while waiting, there are new conflicts to be resolved

**Comment by kguo901** [comment]
I am stuck on understanding why `npm run lint` is failing. This is the log file I am getting.
```
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Users\\kathy\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'lint:spaces' ]
2 info using npm@5.0.3
3 info using node@v6.10.2
4 verbose run-script [ 'prelint:spaces', 'lint:spaces', 'postlint:spaces' ]
5 info lifecycle @~prelint:spaces: @
6 silly lifecycle @~prelint:spaces: no script for prelint:spaces, continuing
7 info lifecycle @~lint:spaces: @
8 verbose lifecycle @~lint:spaces: unsafe-perm in lifecycle true
9 verbose lifecycle @~lint:spaces: PATH: C:\Users\kathy\AppData\Roaming\npm\node_modules\npm\bin\node-gyp-bin;C:\Users\kathy\Documents\GitHub\teammates\node_modules\.bin;C:\Users\kathy\AppData\Roaming\npm\node_modules\npm\bin\node-gyp-bin;C:\Users\kathy\Documents\GitHub\teammates\node_modules\.bin;C:\Users\kathy\AppData\Local\GitHub\PortableGit_f02737a78695063deace08e96d5042710d3e32db\cmd;C:\Users\kathy\AppData\Local\GitHub\PortableGit_f02737a78695063deace08e96d5042710d3e32db\usr\bin;C:\Users\kathy\AppData\Local\GitHub\PortableGit_f02737a78695063deace08e96d5042710d3e32db\usr\share\git-tfs;C:\Users\kathy\AppData\Local\Apps\2.0\G6P3O497.N02\9R058HPX.3OR\gith..tion_317444273a93ac29_0003.0003_5794af8169eeff14;C:\Users\kathy\AppData\Local\GitHub\lfs-amd64_1.5.5;C:\ProgramData\Oracle\Java\javapath;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;c:\Program Files\Intel\WiFi\bin\;c:\Program Files\Common Files\Intel\WirelessCommon\;C:\Program Files\Calibre2\;C:\Program Files (x86)\Skype\Phone\;C:\Program Files\nodejs\;C:\Users\kathy\AppData\Local\Microsoft\WindowsApps;C:\Users\kathy\AppData\Local\atom\bin;C:\Users\kathy\AppData\Roaming\npm;C:\Users\kathy\AppData\Local\Google\Cloud SDK\google-cloud-sdk\bin;C:\Users\kathy\AppData\Local\Microsoft\WindowsApps;;C:\Windows\Microsoft.NET\Framework\v4.0.30319\
10 verbose lifecycle @~lint:spaces: CWD: C:\Users\kathy\Documents\GitHub\teammates
11 silly lifecycle @~lint:spaces: Args: [ '/d /s /c',
11 silly lifecycle   'lintspaces -n -t -d spaces -l 1 -. \'src/**/*.jsp\' \'src/**/*.tag\' \'src/main/**/*.html\' \'src/**/*.xml\' \'src/**/*.json\' \'src/**/*.css\' \'src/**/*.java\' \'src/**/*.es6\' \'src/**/*.properties\' \'*.yml\' \'*.gradle\' \'static-analysis/*.*ml\'' ]
12 silly lifecycle @~lint:spaces: Returned: code: 1  signal: null
13 info lifecycle @~lint:spaces: Failed to exec lint:spaces script
14 verbose stack Error: @ lint:spaces: `lintspaces -n -t -d spaces -l 1 -. 'src/**/*.jsp' 'src/**/*.tag' 'src/main/**/*.html' 'src/**/*.xml' 'src/**/*.json' 'src/**/*.css' 'src/**/*.java' 'src/**/*.es6' 'src/**/*.properties' '*.yml' '*.gradle' 'static-analysis/*.*ml'`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (C:\Users\kathy\AppData\Roaming\npm\node_modules\npm\lib\utils\lifecycle.js:283:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:191:7)
14 verbose stack     at ChildProcess.<anonymous> (C:\Users\kathy\AppData\Roaming\npm\node_modules\npm\lib\utils\spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:191:7)
14 verbose stack     at maybeClose (internal/child_process.js:886:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
15 verbose pkgid @
16 verbose cwd C:\Users\kathy\Documents\GitHub\teammates
17 verbose Windows_NT 10.0.15063
18 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\kathy\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "lint:spaces"
19 verbose node v6.10.2
20 verbose npm  v5.0.3
21 error code ELIFECYCLE
22 error errno 1
23 error @ lint:spaces: `lintspaces -n -t -d spaces -l 1 -. 'src/**/*.jsp' 'src/**/*.tag' 'src/main/**/*.html' 'src/**/*.xml' 'src/**/*.json' 'src/**/*.css' 'src/**/*.java' 'src/**/*.es6' 'src/**/*.properties' '*.yml' '*.gradle' 'static-analysis/*.*ml'`
23 error Exit status 1
24 error Failed at the @ lint:spaces script.
24 error This is probably not a problem with npm. There is likely additional logging output above.
25 verbose exit [ 1, true ]
```

**Comment by whipermr5** [comment]
@kguo901 The Travis build log shows that two files have trailing spaces.

**Comment by thenaesh** [comment]
@wkurniawan07 recently added this tool called lintspaces to our test suite to check for trailing whitespaces. So now these kinds of errors are flagged too. Once you fix them, you should be fine.

**Comment by kguo901** [comment]
@whipermr5 @thenaesh Ok, thanks for your help!

**Comment by damithc** [comment]
@chowyb i'll pass this to you for merge approval. Need to eyeball the resulting page.

**Comment by chowyb** [comment]
Posting this as a comment instead of an actual review due to convenience reasons: the text box for a review is much narrower than the one for a comment =P

Also, you may want to hold off on making any of these changes, as I feel that it may be impractical to always update the UI examples to the current latest version, as due to presently high activity rate on the TEAMMATES repo, many UI changes might slip in between any two iterations of reviews for this PR. For instance, the rubrics UI may soon change if #7415 is merged.
Therefore, I would like to suggest that we "freeze" the current set of UI by not making any additional merges from `master` into this branch, and to only merge in the latest version of `master` at the end, after all reviewers have approved the changes. Any merge conflicts (such as the one present right now) should then be fixed (and reviewed, though "outdated" examples may be accepted at this point), before being merged back into `master`. @damithc would this be alright?

In the following sections, I have listed some differences between the current UI on `instructorHelp`, and the current UI on the actual page, based on the last commit in this branch. I have also listed some contradictions between the UI example and the accompanying text where it happens.

## 1. [Recent Change] Custom Privileges for Instructor
Support for comments on students was removed in #7334, and as such, these checkboxes (in the red boxes) are no longer present.
insthelp1

## 2. Setting Up Questions
There is a contradiction between the text and the UI example; it should be fine to change either the text or the UI so they no longer contradict.
insthelp2a

However, note that the UI example immediately after this follows the original text, so if the text is modified, then this example will have to be changed as well.
insthelp2b

## 3. Setting Up Questions: "another example question"
Same idea as 2., although there is no UI example showing how it looks like from the perspective of a student, so if you change the text, there's no need to worry about other examples contradicting the new text.
insthelp3

## 4. Numerical Scale Question
This example may be out of date. I had originally assumed it was a recent change, however it appears that is unrelated to the issue I had thought about (#7277), which is still open.

Current example in `instructorHelp`:
insthelp4a

Current UI:
insthelp4b

## 5. Distribute Points (among options) Question
Similar to 4.

Current example in `instructorHelp`:
insthelp5a

Current UI:
insthelp5b

## 6. Multiple-choice (single answer) Question
If 4. and 5. are fixed, then the example for MCQ questions will be the only one without the "moderate response" button, which would seem a little out of place.

Current example in `instructorHelp`:
insthelp6

**Comment by chowyb** [comment]
@damithc reminder for this PR, as requested.

**Comment by damithc** [comment]
Sorry @chowyb  I missed this earlier. Do remind me more frequently if I didn't respond in time :-)
I'm even ok to merge this first and fix the minor issues as separate PRs.
Yes, it's going to be a pain to keep these updated with the rapid changes to the UIs. We need to find a more sustainable solution.

**Comment by chowyb** [comment]
@damithc okay 👍 in that case I'll approve this for merging then (although the merge conflict will still need to be resolved, tagging @kguo901 for this).

For keeping the instructor page updated, it was briefly discussed that the use of tags might be possible, although this may require restructuring of the code to accomplish. Regardless, exploring this idea can be left to another issue/PR.

**Pull request merged**